e2e/react-start/selective-ssr/src/routes/posts.$postId.tsx

```diff
@@ -0,0 +1,90 @@
+import { createFileRoute } from '@tanstack/react-router'
+import z from 'zod'
+import { ssrSchema } from '~/search'
+
+export const Route = createFileRoute('/posts/$postId')({
+  validateSearch: z.object({ postId: ssrSchema }),
+  ssr: ({ search }) => {
+    if (typeof window !== 'undefined') {
```

Comment:
Not sure how important this is, but we can probably share the content of the `ssr`, `beforeLoad`, and `component` between the routes yes?